web/web/models/alert.py

```diff
@@ -1,28 +1,235 @@
+                return assigned_to_option
+
+        return False
+
+class ContextType(enum.Enum):
+    TRANSFORMER = 'Transformer'
+    FEEDER = 'Feeder'
+    RETAIL_MARKET = 'Retail market'
+    NONE = 'None'
+
+    @staticmethod
+    def check_value(str_value):
+        '''Takes in string value, returns False if it isn't an accepted enum value, else returns enum type.'''
+
+        for context_type in ContextType:
+            if context_type.value == str_value:
+                return context_type
+
+        return False
+
+class Alert(Model):
+    __tablename__ = 'alerts'
+
+    alert_id = Column(db.Integer, 
+                      primary_key=True,
+                      autoincrement=True, 
+                      nullable=False)
+    
+    alert_type_id = Column(db.Integer, 
+                           db.ForeignKey('alert_types.alert_type_id'), 
+                           nullable=False)
+    
+    assigned_to = Column(db.Enum(AssignedToOptions),  
+                         nullable=False)
+
+    description = Column(db.Text,
+                         nullable=False)
+
+    status = Column(db.Enum(Status),
+                    nullable=False)
+
+    context = Column(db.Enum(ContextType),
+                     nullable=False)
+
+    context_id = Column(db.String(64),
+                        nullable=False)
+
+    resolution = Column(db.Text,
+                        nullable=False)
+    
+    updated_at = Column(TIMESTAMP, 
+                        nullable=False,
+                        server_default=text('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'))
+
+    created_at = Column(TIMESTAMP,
+                        nullable=False,
+                        server_default=func.now())
+
+    # Methods
+    def __repr__(self):
+        return f'<Alert alert_id={self.alert_id} alert_type_id={self.alert_type_id} created_at={self.created_at}>'
+
+    def get_filter_type(self):
+        '''Returns filter type as a string for alert frontend set-up'''
+
+        if self.alert_type.name.value == 'Import Capacity' \
+            or self.alert_type.name.value == 'Export Capacity' \
+                or self.alert_type.name.value == 'Load Yellow' \
+                    or self.alert_type.name.value == 'Load Red':
+            return 'Capacity bounds'
+        
+        if self.alert_type.name.value == 'Price Yellow' \
+            or self.alert_type.name.value == 'Price Red' \
+                or self.alert_type.name.value == 'Price':
+            return 'Price alerts'
+        
+        if self.alert_type.name.value == 'Telecomm':
+            return 'Telecomm alerts'
+        
+        if self.alert_type.name.value == 'Resource':
+            return 'Resource depletion'
+        
+        if self.alert_type.name.value == 'Peak Event':
+            return 'Peak event'
+
+# Relationship declared on other table (dependency on table import for event listener)
+AlertType.alerts = relationship('Alert',
```

Comment:
define on AlertType file.

Reply:
The one caveat is having to access this table in after_insert. I will try moving the after_insert logic into a helper function which should resolve needing this import!